src/compile/selection/transforms/project.ts

```diff
@@ -72,12 +73,20 @@ const project: TransformCompiler = {
       }
     }
 
-    if (init) {
+    if (selDef.init) {
       if (scales.has(selCmpt)) {
         log.warn(log.message.NO_INIT_SCALE_BINDINGS);
       } else {
-        const parseInit = (i: any) => proj.map(p => (i[p.channel] !== undefined ? i[p.channel] : i[p.field]));
-        selCmpt.init = selCmpt.type === 'interval' ? parseInit(init) : array(init).map(parseInit);
+        function parseInit<T extends SelectionInitMapping | SelectionInitArrayMapping>(i: T): T['a'][] {
+          return proj.map(p => (i[p.channel] !== undefined ? i[p.channel] : i[p.field]));
+        }
+
+        if (isIntervalSelection(selDef)) {
+          selCmpt.init = parseInit(selDef.init);
+        } else {
+          const init = isArray(selDef.init) ? selDef.init : [selDef.init];
```

Comment:
We could use array again here once we have https://github.com/vega/vega/pull/1618 in.

Reply:
That relies on the Vega 5 transition to let's do it later.